### PR TITLE
Support for expanded opengraph metadata

### LINKED ISSUE #31: Support "expanded" Open Graph metadata based on og:type
Facebook Oepn Graph defines an [expanded version of embedded metadata](http://ogp.me/#types) depending on the value of `og:type`.

For example:

```
article - Namespace URI: http://ogp.me/ns/article#

    article:published_time - datetime - When the article was first published.
    article:modified_time - datetime - When the article was last changed.
    article:expiration_time - datetime - When the article is out of date after.
    article:author - profile array - Writers of the article.
    article:section - string - A high-level section name. E.g. Technology
    article:tag - string array - Tag words associated with this article.
```

This is used for example on nytimes.com. Snippet:

```
<meta property="og:url" content="http://www.nytimes.com/2016/12/15/arts/music/from-steet-theater-to-wagner-on-the-opera-stage.html" />
<meta property="og:type" content="article" />
<meta property="og:title" content="From Street Theater to Wagner on the Opera Stage" />
<meta property="og:description" content="Àlex Ollé brings an avant-garde sensibility to “The Flying Dutchman,” which he set in Bangladesh instead of Norway. The production opens in Madrid on Saturday." />
<meta property="article:published" itemprop="datePublished" content="2016-12-15T05:55:55-05:00" />
<meta property="article:modified" itemprop="dateModified" content="2016-12-15T06:19:30-05:00" />
<meta property="article:section" itemprop="articleSection" content="Music" />
<meta property="article:section-taxonomy-id" itemprop="articleSection" content="C5BFA7D5-359C-427B-90E6-6B7245A6CDD8" />
<meta property="article:section_url" content="http://www.nytimes.com/section/arts" />
<meta property="article:top-level-section" content="arts" />
<meta property="fb:app_id" content="9869919170" />
```

Currently (as I write these lines version 0.3.0a1) extracts raw `article:...` properties 

```
...
  'article:author': [{'@value': 'http://www.nytimes.com/by/raphael-minder'}],
  'article:collection': [{'@value': 'https://static01.nyt.com/services/json/sectionfronts/arts/music/index.jsonp'}],
  'article:modified': [{'@value': '2016-12-15T06:19:30-05:00'}],
  'article:published': [{'@value': '2016-12-15T05:55:55-05:00'}],
  'article:section': [{'@value': 'Music'}],
  'article:section-taxonomy-id': [{'@value': 'C5BFA7D5-359C-427B-90E6-6B7245A6CDD8'}],
  'article:section_url': [{'@value': 'http://www.nytimes.com/section/arts'}],
  'article:tag': [{'@value': 'Opera'},
                  {'@value': 'Bangladesh'},
                  {'@value': 'Madrid (Spain)'},
                  {'@value': 'Teatro Real'},
                  {'@value': 'Wagner, Richard'}],
  'article:top-level-section': [{'@value': 'arts'}],
  'fb:app_id': [{'@value': '9869919170'}],
  'http://opengraphprotocol.org/schema/description': [{'@value': 'Ã\x80lex '
                                                                 'OllÃ© brings '
                                                                 'an '
                                                                 'avant-garde '
                                                                 'sensibility '
                                                                 'to '
                                                                 'â\x80\x9cThe '
                                                                 'Flying '
                                                                 'Dutchman,â\x80\x9d '
                                                                 'which he set '
                                                                 'in '
                                                                 'Bangladesh '
                                                                 'instead of '
                                                                 'Norway. The '
                                                                 'production '
                                                                 'opens in '
                                                                 'Madrid on '
                                                                 'Saturday.'}],
  'http://opengraphprotocol.org/schema/image': [{'@value': 'https://static01.nyt.com/images/2016/12/16/arts/16ALEXOLLE1-INYT/16ALEXOLLE1-INYT-facebookJumbo.jpg'}],
  'http://opengraphprotocol.org/schema/title': [{'@value': 'From Street '
                                                           'Theater to Wagner '
                                                           'on the Opera '
                                                           'Stage'}],
  'http://opengraphprotocol.org/schema/type': [{'@value': 'article'}],
  'http://opengraphprotocol.org/schema/url': [{'@value': 'http://www.nytimes.com/2016/12/15/arts/music/from-steet-theater-to-wagner-on-the-opera-stage.html'}],
...
```
while they could use the type-dependent OGP namespace

### PR DESCRIPTION
**Pull request by ShivinDass**
Fixes #31 
The merge would result in identifying expanded opengraph metadata properties and replacing them with their corresponding namespaces by adding appropriate prefix to the html tag

### CODE REVIEW

File: extruct/rdfa.py [hunk 1]
```diff
@@ -43,7 +61,28 @@ def extract_items(self, document, base_url=None, expanded=True):
                           vocab_cache_report=False,
                           refresh_vocab_cache=False,
                           check_lite=False)
-
+        document = self.expandedOGSupport(document)
         g = PyRdfa(options, base=base_url).graph_from_DOM(document, graph=Graph(), pgraph=Graph())
         jsonld_string = g.serialize(format='json-ld', auto_compact=not expanded).decode('utf-8')
         return json.loads(jsonld_string)
+
+    def expandedOGSupport(self,document):
```
Comment (Gallaecio) [inline extruct/rdfa.py:69]: :lipstick: I think the new code should use `snake_case`, as the existing code.

File: extruct/rdfa.py [hunk 2]
```diff
@@ -43,7 +61,28 @@ def extract_items(self, document, base_url=None, expanded=True):
...
+      prefixDic = {}
+      for head in document.xpath('//head'):
+        for el in head.xpath('meta[@property and @content]'):
```
Comment (lopuhin) [inline extruct/rdfa.py:72]: @ShivinDass could you please use 4 spaces for indentation, like in the rest of the project?

File: extruct/rdfa.py [hunk 3]
```diff
@@ -43,7 +61,28 @@ def extract_items(self, document, base_url=None, expanded=True):
...
+          prop = el.attrib['property']
+          ns = prop.partition(':')[0]
+          if ns in _OG_NAMESPACES.keys():
+            prefixDic[_OG_NAMESPACES_TAGS[ns]] = _OG_NAMESPACES[ns]
+
+      html_element = None
+      for element in document.iter():
+        if element.tag == 'html':
+          html_element = element
+          break
```
Comment (lopuhin) [inline extruct/rdfa.py:82]: I think using xpath to find the desired element should be shorter.

File: extruct/rdfa.py [hunk 4]
```diff
@@ -26,6 +26,24 @@
     "fb": "http://ogp.me/ns/fb#"
 })
```
Comment (lopuhin) [inline extruct/rdfa.py:27]: @ShivinDass do you think a fix similar to above could be possible here?

Comment (ShivinDass) [reply extruct/rdfa.py:33]: > @ShivinDass do you think a fix similar to above could be possible here?

@lopuhin I took a look at rdflib in more detail and updating the initial_context is a more efficient implementation. I've made the changes and have updated the pull request.

File: extruct/rdfa.py [hunk 5]
```diff
@@ -23,10 +23,15 @@
 
 initial_context["http://www.w3.org/2011/rdfa-context/rdfa-1.1"].ns.update({
     "twitter": "https://dev.twitter.com/cards#",
-    "fb": "http://ogp.me/ns/fb#"
+    "fb": "http://ogp.me/ns/fb#",
+    'og': 'http://ogp.me/ns#',
+	'music': 'http://ogp.me/ns/music#',
```
Comment (lopuhin) [inline extruct/rdfa.py:28]: A minor cosmetic request: could you please make sure indentation is consistent.

File: extruct/rdfa.py [hunk 6]
```diff
@@ -23,10 +23,15 @@
...
+	'video': 'http://ogp.me/ns/video#',
+	'article': 'http://ogp.me/ns/article#',
+	'book': 'http://ogp.me/ns/book#',
+	'profile': 'http://ogp.me/ns/profile#'
 })
 
-
```
Comment (lopuhin) [inline extruct/rdfa.py:29]: A minor cosmetic request: could you please revert unrelated changes (removed lines) here and below - e.g. here two line lines between top-level statements are used, following pep8.

Comment (ShivinDass) [reply extruct/rdfa.py:29]: @lopuhin I'm not too familiar with pep8 but I've made the changes to the best of my understanding. Please do let me know if you find anything else out of order :)

Comment (lopuhin) [reply extruct/rdfa.py:29]: Looks great, thanks @ShivinDass ! By pep-8 I meant https://www.python.org/dev/peps/pep-0008/ but here I mostly wanted to ensure that PR has formatting consistent with current code and no unrelated changes, thank you!
